mdsuite/transformations/map_molecules.py

```diff
@@ -334,6 +354,27 @@ def _load_batch(self, path_list: list, slice: np.s_, factor: list) -> tf.Tensor:
 
         return tf.concat(data, axis=0)
 
+    @staticmethod
+    def _format_batch(batch: dict, scale_factor: np.array) -> tf.Tensor:
+        """
+        Reformat a batch of data into a tensor.
+
+        Parameters
+        ----------
+        batch : dict
+        scale_factor : np.array
+
+        Returns
+        -------
+        data : tf.Tensor
+        """
+        batch.pop(str.encode("data_size"))
+        data = []
+        for i, item in enumerate(batch):
```

Comment:
this relies on the order in which the dictionary is looped over. seems brittle

Reply:
Yes in my list of things to check for why it gives me the wrong values I looked into this. It seems to be working but it is a bit too brittle and I am working on moving it over to dict structures where explicit species are called by name.

Reply:
that sounds better. if both containers have the same type, the looping should be much safer

Reply:
Has been resolved by moving most species operations to act on dictionaries and where not possible, it is ensured that the order a species is called matches that given by the database.